src/test/scala/com/cognite/sdk/scala/v1/DataSetsTest.scala

```diff
@@ -0,0 +1,96 @@
+package com.cognite.sdk.scala.v1
+import java.time.Instant
+
+import com.cognite.sdk.scala.common._
+import com.cognite.sdk.scala.common.SetValue
+
+class DataSetsTest extends SdkTestSpec {
+
+  it should "able to create datasets if doesn't exists" in {
+    val results =
+      client.dataSets.filter(DataSetFilter(externalIdPrefix = Some("testDataset1-"))).compile.toList
+
+    // We can't just keep on creating because datasets doesn't support deletes yet so we filter and then create.
+    // We can have two datasets with same name but not same externalId.
+
+    results.headOption.getOrElse({
+      client.dataSets.create(
+        Seq(
+          DataSetCreate(
+            name = Some("testDataSet-name"),
+            externalId = Some("testDataset1-externalId"))))
+    })
+  }
+
+  it should "support retrieving by ids and throw error for missing id" in {
+    val firstTwoItemIds = client.dataSets.filter(DataSetFilter()).take(2).map(_.id).compile.to.seq
+    assert(firstTwoItemIds.length == 2)
+
+    val maybeItemsRead = client.dataSets.retrieveByIds(firstTwoItemIds)
+    val itemsReadIds = maybeItemsRead.map(_.id)
+
+    assert(itemsReadIds.length == firstTwoItemIds.length)
+    assert(itemsReadIds == firstTwoItemIds)
+
+    assertThrows[CdpApiException] {
+      client.dataSets.retrieveById(1234567890123L)
+    }
+  }
+
+  it should "support retrieving by externalIds and throw error for missing ones" in {
+    val maybeItemsRead = client.dataSets.retrieveByExternalId("testDataset1-externalId")
+
+    assert(maybeItemsRead.externalId == Some("testDataset1-externalId"))
```

Comment:
This can be rewritten as `assert(maybeItemsRead.externalId.contains("testDataset1-externalId"))`.